ekf/tests/test_runner/config_file_handler.c

```diff
@@ -0,0 +1,186 @@
+/*
+ * Phoenix-Pilot
+ *
+ * Part of EKF test runner
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Piotr Nieciecki
+ *
+ * This file is part of Phoenix-Pilot software
+ *
+ * %LICENSE%
+ */
+
+#include "config_file_handler.h"
+
+#include <stdio.h>
+#include <ctype.h>
+#include <stdbool.h>
+#include <errno.h>
+#include <string.h>
+#include <unistd.h>
+
+
+#define EKF_CONFIG_FILE     "etc/ekf.conf"
+#define OLD_EKF_CONFIG_FILE "tmp/old_ekf.conf"
+
+typedef struct {
+	char header[MAX_HEADER_LEN + 1];
+	int headerLen;
+
+	char fieldName[MAX_FIELD_LEN + 1];
+	int fieldNameLen;
+} file_entry_t;
+
+
+int ekftests_restoreConfigFile(void)
+{
+	if (access(EKF_CONFIG_FILE, F_OK) == 0) {
+		if (remove(EKF_CONFIG_FILE) != 0) {
+			return -1;
+		}
+	}
+
+	return rename(OLD_EKF_CONFIG_FILE, EKF_CONFIG_FILE);
+}
+
+
+static inline int ekftests_isWordChar(char c)
+{
+	return (isalnum(c) || c == '_');
+}
+
+
+static int ekftests_fileEntryGet(FILE *file, char **lineBuff, size_t *lineBuffLen, file_entry_t *entry)
+{
+	char *c;
+	int i;
+
+	while (getline(lineBuff, lineBuffLen, file) != -1) {
+		c = *lineBuff;
+
+		while (isspace(*c) != 0) {
+			c++;
+		}
+
+		if (*c == '@') {
+			/* New header was found */
+			c++;
+			i = 0;
+			entry->headerLen = 0;
+
+			/*Header name writing */
+			while (ekftests_isWordChar(*c)) {
+				if (i >= MAX_HEADER_LEN) {
+					fprintf(stderr, "Too long header: %s\n", *lineBuff);
+					errno = EINVAL;
+					return -1;
+				}
+
+				entry->header[i++] = *(c++);
+				entry->headerLen++;
+			}
+
+			entry->header[i] = '\0';
+
+			continue;
+		}
+
+		if (ekftests_isWordChar(*c)) {
+			/* New field was found */
+			entry->fieldNameLen = 0;
+			i = 0;
+
+			do {
+				if (i >= MAX_FIELD_LEN) {
+					fprintf(stderr, "Too long field name: %s\n", *lineBuff);
+					errno = EINVAL;
+					return -1;
+				}
+
+				entry->fieldName[i++] = *(c++);
+				entry->fieldNameLen++;
+			} while (ekftests_isWordChar(*c));
+
+			entry->fieldName[i] = '\0';
+
+			return 0;
+		}
+	}
+
+	return -1;
+}
+
+
+static int ekftests_fieldToChange(const file_entry_t *entry, const char *fieldToChange)
+{
+	if (strncmp(entry->header, fieldToChange, entry->headerLen) != 0) {
+		return -1;
+	}
+
+	if (fieldToChange[entry->headerLen] != '/') {
+		return -1;
+	}
+
+	return strncmp(entry->fieldName, fieldToChange + entry->headerLen + 1, entry->fieldNameLen);
+}
+
+
+int ekftests_prepareConfigFile(char **fieldsToChange, int fieldsToChangeCnt)
```

Comment:
```suggestion
int ekftests_prepareConfigFile(char **fields, int fieldsCnt)
```
phrase `fieldsToChange` is used extensively. Please add description to this function with argument description.